pkg/tinydtls/contrib/sock_dtls.c

```diff
@@ -296,6 +392,67 @@ int sock_dtls_create(sock_dtls_t *sock, sock_udp_t *udp_sock,
     return 0;
 }
 
+int sock_dtls_set_psk_identity_hint(sock_dtls_t *sock, const char *hint)
+{
+    assert(sock);
+    if (strlen(hint) > CONFIG_DTLS_PSK_ID_HINT_MAX_SIZE) {
+        DEBUG("sock_dtls: could not set hint due to buffer size\n");
+        return -1;
+    }
+    strcpy(sock->psk_hint, hint);
+    return 0;
+}
+
+int sock_dtls_add_credential(sock_dtls_t *sock, credman_tag_t tag)
```

Comment:
I'm thinking if we need to call `sock_dtls_add_credential` for each new `credman_add` anyways, would it be better to include `credman_add` in `sock_dtls_add_credential` also? The parameter list might need to change then though.

Reply:
It is completely legitimate to use credentials for multiple sockets. Running `credman_add()` every time would mean that every time you reuse a credential, you would first have to find out that the credential already exists.